src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleCloudStorageDAO.scala

```diff
@@ -58,6 +67,36 @@ class HttpGoogleCloudStorageDAO(clientSecretsJson: String, dataStoreFactory: Dat
       .execute()
   }
 
+  override def deleteBucket(userId: String, projectId: String, bucketName: String): Unit = {
+    import system.dispatcher
+    def bucketDeleteFn(): Unit = {
+      val credential = getCredential(userId)
+      val storage = new Storage.Builder(httpTransport, jsonFactory, credential).build()
+      val bucket: Bucket = storage.buckets().get(bucketName).execute()
+
+      //Google doesn't let you delete buckets that are full.
+      //You can either remove all the objects manually, or you can set up lifecycle management on the bucket.
+      //This can be used to auto-delete all objects next time the Google lifecycle manager runs (~every 24h).
+      //More info: http://bit.ly/1WCYhhf and http://bit.ly/1Py6b6O
+      val deleteEverythingRule = new Lifecycle.Rule()
+        .setAction(new Action().setType("delete"))
+        .setCondition(new Condition().setAge(0))
+
+      val lifecycle = new Lifecycle().setRule(List(deleteEverythingRule))
+      bucket.setLifecycle(lifecycle)
+
+      Try(storage.buckets().delete(bucketName).execute) match {
+        //Google returns 409 Conflict if the bucket isn't empty.
+        case Failure(gjre: GoogleJsonResponseException) if gjre.getDetails.getCode == 409 =>
+          system.scheduler.scheduleOnce(deletedBucketCheckSeconds seconds)(bucketDeleteFn)
+        //TODO: I am entirely unsure what other cases might want to be handled here.
+        //404 means the bucket is already gone, which is fine.
+        //Success is also fine, clearly.
+      }
+    }
+    system.scheduler.scheduleOnce(deletedBucketCheckSeconds seconds)(bucketDeleteFn)
```

Comment:
The server restart issue is present across the codebase and has a ticket tracking it in DSDEEPB-1075 .

I was planning on returning 203 Accepted for delete workspace, which is the response for "I've got your request and will finish doing it later". I was also planning on adding a message saying "your bucket should be gone in 24h". Beyond keeping the connection open as we iterate over and delete every single file in the bucket (possibly for hours) I don't think there's much else we can do.